mmcv/runner/ipu/hooks.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from mmcv.utils import TORCH_VERSION, digit_version
+from mmcv.runner.hooks import HOOKS, OptimizerHook, LrUpdaterHook
+
+
+def wrap_lr_update_hook(lr_hook_class):
+    """A wrapper function to wrap any subclass of LrUpdaterHook
+    IPU needs extra operations to upload optimizer setting,
+    this wrapper will override function(_set_lr) of a subclass of
+    LrUpdaterHook.
+    """
+    assert issubclass(lr_hook_class, LrUpdaterHook)
+
+    class ipu_lr_hook_class(lr_hook_class):
+        def _set_lr(self, runner, *args, **kwargs):
+            result = super()._set_lr(runner, *args, **kwargs)
+            assert result is None  # _set_lr should return nothing
+            runner.model.setOptimizer(runner.optimizer)
+    return ipu_lr_hook_class
+
+
+def wrap_optimizer_hook(optimizer_hook_class,):
+    """A wrapper function to wrap OptimizerHook
+    This is an non-intrusive implementation of wrapping optimizer hook
+    (or you need to change every config file to use IPU optimizer hook)
+    IPU's clip-norm implementation is different from pytorch, so there
+    should be an error raised when using clip-norm.
+    """
+    assert optimizer_hook_class == OptimizerHook,\
```

Comment:
issubclass instead of ==?

Reply:
only support OptimizerHook, currently not support subclass of OptimizerHook like GradientCumulativeOptimizerHook